### PR TITLE
8275887: jarsigner prints invalid digest/signature algorithm warnings if keysize is weak/disabled

### PR DESCRIPTION
**Pull request by GoeLin**
I backport this change from 17u because I had to do adaptions for 17 that 
are needed here, too.  I needed further fixes, though:

src/java.base/share/classes/sun/security/pkcs/SignerInfo.java
Slight difference in context, change applies to code clean.
I had to change a record to a class to make the change compile
with Java 11.

src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java
Chunks don't apply because in 11 method JAR_DISABLED_CHECK 
lacks the prefix JAR_.
checkWeakKey() and  checkWeakAlg() are not implemented in 11. Hunk omitted.

test/jdk/sun/security/tools/jarsigner/TimestampCheck.java
Resolved adding bugID to @bug.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8275887](https://bugs.openjdk.org/browse/JDK-8275887): jarsigner prints invalid digest/signature algorithm warnings if keysize is weak/disabled

### Reviewers
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**) ⚠️ Review applies to [e65205a9](https://git.openjdk.org/jdk11u/pull/56/files/e65205a9b6eb8a0a547659036a7338ce0af9c530)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u pull/56/head:pull/56` \
`$ git checkout pull/56`

Update a local copy of the PR: \
`$ git checkout pull/56` \
`$ git pull https://git.openjdk.org/jdk11u pull/56/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 56`

View PR using the GUI difftool: \
`$ git pr show -t 56`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u/pull/56.diff">https://git.openjdk.org/jdk11u/pull/56.diff</a>

</details>

### CODE REVIEW

File: src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java [hunk 1]
```diff
@@ -973,20 +973,25 @@ void verifyJar(String jarName)
...
                                 JarConstraintsParameters jcp =
-                                    new JarConstraintsParameters(chain, si.getTimestamp());
+                                    new JarConstraintsParameters(chain, tsDate);
+                                JarConstraintsParameters jcpts =
+                                    new JarConstraintsParameters(
+                                        tsSi.getCertificateChain(tsToken),
+                                        tsDate);
                                 history = String.format(
-                                        rb.getString("history.with.ts"),
+                                        rb.getString("history.with.ts"),      // GLGLGL
```
Comment (TheRealMDoerr) [inline src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java:985]: I suggest to remove the comment.

**Review by TheRealMDoerr — APPROVED** [review]
See below. Otherwise, LGTM. Thanks for backporting.

File: src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java (hunk 1 above)
Comment (GoeLin) [reply src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java]: Removed. Thanks for finding this!

**Comment by GoeLin** [comment]
/integrate

Thanks for reviewing!

**Pull request closed without merging**